client/rest-high-level/src/main/java/org/elasticsearch/client/SecurityClient.java

```diff
@@ -92,6 +94,33 @@ public void putUserAsync(PutUserRequest request, RequestOptions options, ActionL
             PutUserResponse::fromXContent, listener, emptySet());
     }
 
+    /**
+     * Removes user from the native realm synchronously.
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-delete-user.html">
+     * the docs</a> for more.
+     * @param request the request with the user to delete
+     * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
+     * @return the response from the delete user call
+     * @throws IOException in case there is a problem sending the request or parsing back the response
+     */
+    public DeleteUserResponse deleteUser(DeleteUserRequest request, RequestOptions options) throws IOException {
```

Comment:
There is currently a PR for adding `Optional<Response>` performRequest methods, so you can, instead of having a `isFound` boolean inside your `Response`, return an `Optional.empty()`. I am not sure it would work for these APIs tho, as the docs did not specify them throwing a 404 if the user is not found. The code reads like the 404 would be thrown if the user is not found, but the API docs specifies it returns found=false. Can you verify which one is correct, and if return=false is correct we should probably remove the `singleton(404)` in the ignores parameter.

Ill leave this here to look at, for returning Optional.empty() on a 404 not found, in case you find it useful. #35166

Reply:
It will throw a 404 if the user is not found ant that is the reason added `singleton(404)` as ignore parameter (my first approach was without it). I would leave it like that to be consistent with other actions on this package (see `deleteRole`). 

If we want to return an `Optional<Response>` then it should be done to all the equivalent requests. WDYT?

Reply:
BTW just try to delete a non existing user on a local installation and it returns false. So I guess thus behaviour is the expected one.

Reply:
so then the 404 does not actually happen, right? If so we should remove it. Im also all for using Optional instead of found=false, in general, but you dont have to go fix that all right now.

Reply:
No, the 404 actually happen. So this is needed to return false.

Reply:
cool, so we can remove it from the ignores list

Reply:
Not sure if I am following this right. The call should not throw an error, either we implement the Optional and it will be empty or we go for the current implementation. Removing the 404 from the ignore list result in an exception being thrown.

Reply:
Sorry I misread your previous statement about the 404 actually happening as "the 404 does not happen" ... This was my mistake! 

> If we want to return an Optional<Response> then it should be done to all the equivalent requests.

Good point! @jaymode we should consider using this Optional here instead of found:false return in the response. the correct return from the HLRC is Optional for 404's, but this is different in the other APIs. Id prefer to do the Optional work here, now, rather than have another API that needs migrating. WDYT?

Reply:
64e5c25 added support for this.